### PR TITLE
[docs] add document conditional-functions

### PR DESCRIPTION
**Pull request by steadyBoy**
Add conditional-functions

# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Add conditional-functions.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/sql-reference/sql-functions/conditional-functions/if.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+# if
+## description
+### Syntax
+
+`if(boolean condition, type valueTrue, type valueFalseOrNull)`
+
```
Comment (dataroaring) [inline docs/en/sql-reference/sql-functions/conditional-functions/if.md:32]: I am confused by 'type' here. What is meaning of 'type'?

File: docs/en/sql-reference/sql-functions/conditional-functions/ifnull.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# ifnull
+## description
+### Syntax
+
+`ifnull(expr1, expr2)`
+
+
+If the value of expr1 is not null, V1 is returned, otherwise expr2 is returned
+
```
Comment (dataroaring) [inline docs/en/sql-reference/sql-functions/conditional-functions/ifnull.md:35]: V1 here is a typo?

File: docs/zh-CN/sql-reference/sql-functions/conditional-functions/case.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# case
+## description
+### Syntax
+
+`CASE a WHEN b THEN c [WHEN d THEN e]... [ELSE f] END`
+
+OR
+
+`CASE WHEN a THEN b [WHEN c THEN d]* [ELSE e] END`
+
```
Comment (dataroaring) [inline docs/zh-CN/sql-reference/sql-functions/conditional-functions/case.md:36]: It is hard to understand a-e,  we need word like condition to express more things.

**Review by hf200012 — DISMISSED** [review]
LGTM

File: docs/en/sql-reference/sql-functions/conditional-functions/ifnull.md (hunk 2 above)
Comment (steadyBoy) [reply docs/en/sql-reference/sql-functions/conditional-functions/ifnull.md:35]: Yes, I'll fix it

File: docs/zh-CN/sql-reference/sql-functions/conditional-functions/case.md (hunk 3 above)
Comment (steadyBoy) [reply docs/zh-CN/sql-reference/sql-functions/conditional-functions/case.md:49]: OK

File: docs/en/sql-reference/sql-functions/conditional-functions/if.md (hunk 1 above)
Comment (steadyBoy) [reply docs/en/sql-reference/sql-functions/conditional-functions/if.md:32]: I added in the description

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**